referral-form-data-process/main.ts

```diff
@@ -4,22 +4,27 @@ import { generateAuth } from "./lib/generateGoogleAuth";
 import { getDataFromS3 } from "./lib/getDataFromS3";
 
 export const handler = async (sqsEvent: SQSEvent) => {
-  const clientEmail = process.env.CLIENTEMAIL as string;
-  const privateKey = process.env.PRIVATEKEY as string;
-  const templateDocumentId = process.env.TEMPLATEDOCUMENTID as string;
+  const clientEmail = process.env.CLIENT_EMAIL as string;
+  const privateKey = process.env.PRIVATE_KEY as string;
+  const templateDocumentId = process.env.TEMPLATE_DOCUMENT_ID as string;
   const title = process.env.TITLE as string;
+  const formattedPrivateKey = privateKey.replace(/\\n/g, "\n");
 
   const formDataObjects = await getDataFromS3(sqsEvent);
+  const googleAuthToken = await generateAuth(clientEmail, formattedPrivateKey);
 
-  const googleAuthToken = await generateAuth(clientEmail, privateKey);
-
-  formDataObjects.forEach(
-    async (formData) =>
-      await createDocumentFromTemplate(
-        googleAuthToken,
-        templateDocumentId,
-        title,
-        formData
-      )
+  await Promise.all(
+    formDataObjects.map(async (formData) => {
+      await new Promise((resolve) => {
+        resolve(
+          createDocumentFromTemplate(
+            googleAuthToken,
+            templateDocumentId,
+            title,
+            formData
+          )
+        );
+      });
+    })
```

Comment:
I _believe_ these two are equivalent, and it's a little more concise:

```suggestion
  await Promise.all(
    formDataObjects.map((formData) =>
      createDocumentFromTemplate(
        googleAuthToken,
        templateDocumentId,
        title,
        formData
      )
    )
  );
```

Reply:
Tested this and it works 🎉 . Done in f677781